ts/Series/DataModifyComposition.ts

```diff
@@ -0,0 +1,788 @@
+        mode: 'compare'|'cumulative',
+        modeState?: boolean|null|string,
+        redraw?: boolean
+    ): void {
+        if (!this.isXAxis) {
+            this.series.forEach(function (series): void {
+                if (
+                    mode === 'compare' &&
+                    (
+                        isString(modeState) ||
+                        typeof modeState === 'undefined' ||
+                        modeState === null
+                    )
+                ) {
+                    series.setCompare(modeState, false);
+                } else if (
+                    mode === 'cumulative' &&
+                    (
+                        typeof modeState === 'boolean' ||
+                        typeof modeState === 'undefined' ||
+                        modeState === null
+                    )
+                ) {
+                    series.setCumulative(modeState, false);
+                }
+            });
+
+            if (pick(redraw, true)) {
+                this.chart.redraw();
+            }
+        }
+    }
+
+    /**
+     * Extend the tooltip formatter by adding support for the point.change
+     * variable as well as the changeDecimals option.
+     *
+     * @ignore
+     * @function Highcharts.Point#tooltipFormatter
+     *
+     * @param {string} pointFormat
+     */
+    function tooltipFormatter(this: Point, pointFormat: string): string {
+        const point: any = this,
+            { numberFormatter } = point.series.chart,
+            replace = function (value: string): void {
+                pointFormat = pointFormat.replace(
+                    '{point.' + value + '}',
+                    (point[value] > 0 && value === 'change' ? '+' : '') +
+                        numberFormatter(
+                            point[value],
+                            pick(point.series.tooltipOptions.changeDecimals, 2)
+                        )
+                );
+            };
+
+        if (defined(point.change)) {
+            replace('change');
+        }
+
+        if (defined(point.cumulativeSum)) {
+            replace('cumulativeSum');
+        }
+
+        return pointTooltipFormatter.apply(this, [pointFormat]);
+    }
+
+    /**
+     * Extend series.init by adding a methods to modify the y values used
+     * for plotting on the y axis. For compare mode, this method is called both
+     * from the axis when finding dataMin and dataMax,
+     * and from the series.translate method.
+     *
+     * @ignore
+     * @function Highcharts.Series#init
+     */
+    function afterInit(this: Series): void {
+        this.dataModify = new Additions(this as SeriesComposition);
+
+        if (this.options.compare) {
+            // Set comparison mode
+            this.dataModify.initCompare(this.options.compare);
+        } else if (this.options.cumulative) {
+            // Set Cumulative Sum mode
+            this.dataModify.initCumulative(this.options.cumulative);
+        }
+    }
```

Comment:
I am wondering if we should set `this.dataModify` only when we have an active modifier, and otherwise delete it. Then in Series.ts we can test for `series.dataModify`, which would always have a `modifyValue`. Instead of a deep test for `series.dataModify && series.dataModify.modifyValue` which we are currently doing repeatedly. By implication, we would not set `modifyValue` to null, but instead do something like this:

```ts

        let dataModify: Additions|undefined;
        if (this.options.compare || this.options.cumulative) {
            dataModify = new Additions(this as SeriesComposition);
            if (this.options.compare) {
                // Set comparison mode
                dataModify.initCompare(this.options.compare);
            } else {
                // Set Cumulative Sum mode
                dataModify.initCumulative(this.options.cumulative);
            }
        }
        this.dataModify = dataModify;
```
This pattern is more in line with other compositions we have. Like for example `Axis.logarithmic`- when `logarithmic` is set, we can safely use `logarithmic.getLogTickPositions`.